components/generic/tabs/components/tabs.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { memo } from 'react'
+import cx from 'classnames'
+
+import { Tab } from '../types'
+
+export interface Props {
+  tabs: Tab[]
+  value: number | string
+  onChange: (event: React.ChangeEvent<{}>, value: number | string) => void
+  className?: string
+}
+
+const Tabs: React.FC<Props> = ({ tabs, value, onChange, className }) => {
+  const show = tabs.find((tab) => tab.id === value && tab.show !== false)
+
+  return (
+    <>
+      {show && (
+        <>
+          <ul className={cx('navigation', className)}>
+            {tabs.map(
+              ({ id, label, counter, disabled = false, show = true }) =>
+                show && (
+                  <li
+                    onClick={(e) => onChange(e, id)}
+                    key={id}
+                    className={cx({ active: value === id, disabled })}
+                    value={id}
+                  >
+                    <a>
+                      <span>{label}</span>
+                      {counter ? <span>({counter})</span> : null}
```

Comment:
maybe instead of label and counter, can we leave more abstract and let the "tab title" be any react component.

Reply:
but anyway.. Don't change it, i like it